src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -859,7 +859,27 @@ impl<'tcx> IntRange<'tcx> {
             }
             ConstantValue(val) if is_integral(val.ty) => {
                 let ty = val.ty;
-                if let Some(val) = val.try_eval_bits(tcx, param_env, ty) {
+                let size = match ty.sty {
```

Comment:
Please move this optimization into the `try_eval_bits` function, as all of rustc could benefit from it